### PR TITLE
enable thermo calculation and save to h5

### PR DESCRIPTION
**Pull request by liuyu-chem1996**

### CODE REVIEW

File: gpu4pyscf/drivers/dft_driver.py [hunk 1]
```diff
@@ -214,9 +217,27 @@ def run_dft(mol_name, config, charge=None, spin=0):
             start_time = time.time()
             h = mf.Hessian()
             h.auxbasis_response = 2
-            h_dft = h.kernel().transpose([0,2,1,3]).reshape([3*natm, 3*natm])
+            _h_dft = h.kernel()
```
Comment (liuyu-chem1996) [inline gpu4pyscf/drivers/dft_driver.py:220]: 220-224 recover back

File: gpu4pyscf/drivers/dft_driver.py [hunk 2]
```diff
@@ -226,6 +247,26 @@ def run_dft(mol_name, config, charge=None, spin=0):
         h5f.create_dataset('hess', data=h_dft)
         h5f.create_dataset('hess_time', data=hess_time)
 
+        if with_thermo:
```
Comment (liuyu-chem1996) [inline gpu4pyscf/drivers/dft_driver.py:250]: alignment

File: gpu4pyscf/drivers/dft_driver.py [hunk 3]
```diff
@@ -88,6 +90,7 @@ def run_dft(mol_name, config, charge=None, spin=0):
     with_solvent   = config.get('with_solvent',   False)
     with_grad      = config.get('with_grad',      True)
     with_hess      = config.get('with_hess',      True)
+    with_thermo    = config.get('with_thermo',    True)
```
Comment (liuyu-chem1996) [inline gpu4pyscf/drivers/dft_driver.py:93]: false

File: gpu4pyscf/drivers/dft_driver.py [hunk 4]
```diff
@@ -214,9 +217,27 @@ def run_dft(mol_name, config, charge=None, spin=0):
...
+            h_diag = h_dft.sum(axis=0)
+            h_dft -= np.diag(h_diag)
+            _h_dft = h_dft.reshape([natm, 3, natm, 3]).transpose([0, 2, 1, 3])
             hess_time = time.time() - start_time
             print(f'compute time for hessian: {hess_time:.3f} s')
+
+            if with_thermo:
+                # harmonic analysis
+                start_time = time.time()
+                normal_mode = thermo.harmonic_analysis(mol, _h_dft)
```
Comment (liuyu-chem1996) [inline gpu4pyscf/drivers/dft_driver.py:231]: h_dft

File: gpu4pyscf/drivers/dft_driver.py [hunk 5]
```diff
@@ -231,6 +233,20 @@ def run_dft(mol_name, config, charge=None, spin=0):
             h_dft = h.kernel().transpose([0,2,1,3]).reshape([3*natm, 3*natm])
             hess_time = time.time() - start_time
             print(f'compute time for hessian: {hess_time:.3f} s')
+
+            if with_thermo:
+                # harmonic analysis
+                start_time = time.time()
+                normal_mode = thermo.harmonic_analysis(mol, h.kernel())
```
Comment (wxj6000) [inline gpu4pyscf/drivers/dft_driver.py:240]: @liuyu-chem1996 `h.kernel()` is executed twice. It is expensive. Simply reuse `h_dft`.

File: gpu4pyscf/drivers/dft_driver.py [hunk 6]
```diff
@@ -228,9 +230,24 @@ def run_dft(mol_name, config, charge=None, spin=0):
...
-            h_dft = h.kernel().transpose([0,2,1,3]).reshape([3*natm, 3*natm])
+            _h_dft = h.kernel()
+            h_dft = _h_dft.transpose([0,2,1,3]).reshape([3*natm, 3*natm])
             hess_time = time.time() - start_time
             print(f'compute time for hessian: {hess_time:.3f} s')
+
+            if with_thermo:
+                # harmonic analysis
+                start_time = time.time()
+                normal_mode = thermo.harmonic_analysis(mol, _h_dft)
```
Comment (wxj6000) [inline gpu4pyscf/drivers/dft_driver.py:241]: `h_dft` not `_h_dft`

**Pull request merged**